packages/firestore/src/core/query.ts

```diff
@@ -199,6 +199,22 @@ export class Query {
     );
   }
 
+  /**
+   * Returns true if this query does not specify any query constraints that
+   * could remove results.
+   */
+  matchesAllDocuments(): boolean {
+    return (
+      this.filters.length === 0 &&
+      this.limit === null &&
+      this.startAt == null &&
+      this.endAt == null &&
+      (this.explicitOrderBy.length === 0 ||
+        (this.explicitOrderBy.length === 1 &&
+          this.getFirstOrderByField()!.isKeyField()))
```

Comment:
You could do `this.explicitOrderBy[0].field.isKeyField()` which would be more congruous with the previous check and avoid the need for `!`.

Reply:
Done. Thanks for the suggestion.